buildkite/resource_team_member_test.go

```diff
@@ -0,0 +1,238 @@
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTeamMemberResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTeamMemberConfigBasic("MEMBER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team member has the correct values in Buildkite's system
+					testAccCheckTeamMemberRemoteValues(&resourceTeamMember, "MEMBER"),
+					// Confirm the team member has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_team_member.test", "role", "MEMBER"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTeamMember_add_remove_non_default_role(t *testing.T) {
+	var resourceTeamMember TeamMemberNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTeamMemberResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTeamMemberConfigBasic("MAINTAINER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team member has the correct values in Buildkite's system
+					testAccCheckTeamMemberRemoteValues(&resourceTeamMember, "MAINTAINER"),
+					// Confirm the team member has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_team_member.test", "role", "MAINTAINER"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTeamMember_update(t *testing.T) {
+	var resourceTeamMember TeamMemberNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTeamMemberResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTeamMemberConfigBasic("MEMBER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team has the correct values in Buildkite's system
+					testAccCheckTeamMemberRemoteValues(&resourceTeamMember, "MEMBER"),
+				),
+			},
+			{
+				Config: testAccTeamMemberConfigBasic("MAINTAINER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team has the correct values in Buildkite's system
+					testAccCheckTeamMemberRemoteValues(&resourceTeamMember, "MAINTAINER"),
+				),
+			},
+		},
+	})
+}
+
+// Confirm that this resource can be imported
+func TestAccTeamMember_import(t *testing.T) {
```

Comment:
I assume when importing that the user should provide the graphql ID of the team membership record? If so, how would they find that to paste into the import cli command?

Reply:
Yeah thats right, it's not displayed in the web anywhere like some other things. It'll need to be found through a GraphQL query. I'll create a ticket to follow up with adding an example query to the docs for people to use when importing resources

Reply:
The docs for this resource contain an example graphql query to help with this